### PR TITLE
Attachment finetuning

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
- [Add] attachment functionality
- [Fix] Create requirement without existing RequirementsSpecification

### CODE REVIEW

File: CDP4Common/AutoGenHelpers/DefaultPermissionProvider.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public class DefaultPermissionProvider : IDefaultPermissionProvider
             { ClassKind.ArchitectureElement, ParticipantAccessRightKind.SAME_AS_SUPERCLASS },
             { ClassKind.ArrayParameterType, ParticipantAccessRightKind.SAME_AS_SUPERCLASS },
             { ClassKind.Attachment, ParticipantAccessRightKind.SAME_AS_CONTAINER },
-            { ClassKind.Behavior, ParticipantAccessRightKind.SAME_AS_SUPERCLASS },
+            { ClassKind.Behavior, ParticipantAccessRightKind.SAME_AS_CONTAINER },
```
Comment (alexatrhea) [inline CDP4Common/AutoGenHelpers/DefaultPermissionProvider.cs:58]: Hmm Ive already landed this on the GCD branch. Not sure why it is being diffed here.

https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/commit/17df2bfe33a617fe3f4bbd3bfc515a4a150a59e3

File: CDP4Common/Dto/Attachment.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// --------------------------------------------------------------------------------------------------------------------
```
Comment (alexatrhea) [inline CDP4Common/Dto/Attachment.cs:1]: Again im confused....

https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/commit/13bc764c631573967cb7427de1b664d783040bbf

**Review by alexatrhea — CHANGES_REQUESTED** [review]
Something weird with this diff. I have the impression I have made some of these fixes already so how does it able to to diff it in this way. Have you rebased?

File: CDP4Dal/Operations/ThingTransaction.cs [hunk 3]
```diff
@@ -617,14 +617,14 @@ public OperationContainer FinalizeTransaction()
...
+        /// Get all the files that need to be added to the DataStore according to the added <see cref="Thing"/>s that implement <see cref="ILocalFile"/>.
         /// </summary>
         /// <returns>An Array of strings that contain the local paths in the context of the users' computer</returns>
         public string[] GetFiles()
         {
             var files = new List<string>();
-
-            foreach (var thing in this.AddedThing.OfType<ILocalFile>().Union(this.UpdatedThing.OfType<ILocalFile>()).Where(x => x.LocalPath != null))
+            
+            foreach (var thing in this.AddedThing.Where(x => x is ILocalFile).Cast<ILocalFile>().Where(x => x.LocalPath != null))
```
Comment (SimonWoodRhea) [inline CDP4Dal/Operations/ThingTransaction.cs:627]: what about just OfType<ILocalFile>() instead of a where and a cast?

Comment (lxatrhea) [reply CDP4Dal/Operations/ThingTransaction.cs:627]: ReSharper suggested it, probably during refactoring when the check on LocalPath (for which the Cast was needed) wasn't there temporarily.
Fixed it.

File: CDP4Common/Dto/Attachment.cs [hunk 4]
```diff
@@ -1,23 +1,23 @@
 ﻿// --------------------------------------------------------------------------------------------------------------------
 // <copyright file="Attachment.cs" company="RHEA System S.A.">
 //    Copyright (c) 2015-2021 RHEA System S.A.
-// 
-//    Author: Sam Gerené, Alex Vorobiev, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Nathanael Smiechowski, Ahmed Ahmed, Simon Wood
-// 
+//
```
Comment (alexatrhea) [inline CDP4Common/Dto/Attachment.cs:4]: looks like you have old header file def

**Review by alexatrhea — APPROVED** [review]

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**